File: frontend/src/app/services/user.service.ts
import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { User } from '../models/User';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  private apiUrl = 'http://localhost:8080/user';
  constructor(private http: HttpClient) {} 
  login(email: string, password: string ): Observable<any> {
    const params = new HttpParams()
      .set('email', email)
      .set('password', password);
    const urlWithParams = `${this.apiUrl}/signIn?${params.toString()}`;

      // Effectuer la requête POST avec les paramètres dans l'URL
      return this.http.post<any>(urlWithParams, {});
  }
  register(user: User ): Observable<any> {
      return this.http.post<any>(`${this.apiUrl}/signUp`, user, {
      headers: new HttpHeaders({
        'Content-Type': 'application/json'
      })})
  }
  updateUser(user: User ): Observable<any> {
    return this.http.put<any>(`${this.apiUrl}/updateUser`, user, {
    headers: new HttpHeaders({
      'Content-Type': 'application/json'
    })})
  }
  deleteUser(id : string): Observable<any> {
    return this.http.delete<any>(`${this.apiUrl}/deleteUser/${id}`);
  }
  getUserById(id : string): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/getUserById/${id}`);
  }
  searchForUser(firstName : string , lastName : string): Observable<any> {
    const params = new HttpParams()
      .set('firstName', firstName)
      .set('lastName', lastName);
    return this.http.get<any>(`${this.apiUrl}/getUserByName`,{params});
  }
  
}

File: backend/src/main/java/com/example/backend/Service/UserService.java
package com.example.backend.Service;

import com.example.backend.Entity.Review;
import com.example.backend.Entity.User;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;


import java.util.List;

public interface UserService {
    ResponseEntity<User> signIn(String email , String password);
    ResponseEntity<List<User>> getUserByName(String firstName ,String lastName);
    ResponseEntity<User> getUserById(Long idUser);
    ResponseEntity<List<User>> getAllUsers();
    ResponseEntity<?> signUp(User user);
    ResponseEntity<User> updateUser(User user);
    ResponseEntity<String> deleteUser(Long idUser);
    
}
